### ISSUE TITLE
Change the code to make regression

### ISSUE THREAD

**Issue by jordangierschendorf** [kratzert/finetune_alexnet_with_tensorflow#20]
Dear kratzert,
First, thanks for your amazing work !
I would like to use your code to solve a regression problem, do have some time to help me ? 
Best.

**Comment by kratzert** [comment]
To switch from classification to regression you just have to discard the softmax layer at the very end of the network and work with the pure activations. You also need to change the the loss function you like to minimize, since categorical loss does not make sense for regression. You can use something like rmse, mse etc.

**Comment by jordangierschendorf** [comment]
Thanks for your answer.

I understand. For my regression I need to have only one output (my issue deals with age estimation). So the best way is to switch directly self.NUM_CLASSES = 1 in fc8 with relu=True ? 

Furthermore, I would like to inject my label values instead of one_hot label in input. Do you have quick tips to do it ?

**Comment by kratzert** [comment]
1 output node is correct and no relu should be applied, since this brings non linearity to your model output. you want to normalize your output labels also, so that negative "ages" are possible model outputs (helps you model to train) and then do back transformation on this value.

And for your second question: just adapt the dataset api mapping function by removing the one_hot function and pass you true labels/values without changes to the return. 

My general tip: None of this is a really hard problem and it seems that you don't have so much knowledge on ML/DL. Maybe you should do any of the online freely available online courses to learn the basics (e.g. Andrew Ng Coursera course on ML). There you learn basically everything you need for this project but on simpler models. Using any ConvNet then instead of a simple model doesn't change much.
